packages/validator/src/services/indices.ts

```diff
@@ -83,12 +83,15 @@ export class IndicesService {
       return this.pollValidatorIndicesPromise;
     }
 
-    this.pollValidatorIndicesPromise = this.pollValidatorIndicesInternal(pubkeysHex);
     // Once the pollValidatorIndicesInternal() resolves or rejects null the cached promise so it can be called again.
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.pollValidatorIndicesPromise.finally(() => {
-      this.pollValidatorIndicesPromise = null;
-    });
+    this.pollValidatorIndicesPromise = this.pollValidatorIndicesInternal(pubkeysHex)
+      .catch((e) => {
+        this.logger.warn("Failed to get indices for pending pubkeys", {}, e);
+        return [];
```

Comment:
After reviewing where `pollValidatorIndices` I don't think it is correct to catch the error here, the function returns the promise and it's the responsibility of the upstream code to handle it. Reviewing all callers, it seems like we handle all errors

e.g. in attestation duties service
https://github.com/ChainSafe/lodestar/blob/f29a6db05624fc454e586790e4f977ec0817ed25/packages/validator/src/services/attestationDuties.ts#L151-L156

I don't see how we got the uncaught exception in PR description, there must be some place were this is not correctly handled, the caller function is not part of stack trace which is strange

Reply:
the current solution fixes the uncaught exception, you are free to update to a better solution if you have one

Reply:
If we catch the error here we will lose the log details from the upstream error handlers which include more info in the message of where polling indices failed.

Catching the error might also reduce security of our doppelganger protection as we would no longer throw an error anymore, meaning some indices could be missed in some edge cases
https://github.com/ChainSafe/lodestar/blob/f29a6db05624fc454e586790e4f977ec0817ed25/packages/validator/src/services/doppelgangerService.ts#L143

Need to figure our where this uncaught exception is coming from and handle it there correctly, I am not sure why the caller is not part of the stack trace